internal/filesystem/filesystem.go

```diff
@@ -1,31 +1,164 @@
 package filesystem
 
 import (
+	"context"
+	"errors"
+	"fmt"
 	"io"
 	"io/fs"
+	"os"
+	"time"
 )
 
-// TODO: name?
-type FS interface {
+type (
+	// TODO: [review] These interfaces come from our pre fs.FS implementations
+	// Things like names and patterns may (need to) change.
+	// Some things might need to be added as well.
+	IDFS interface {
+		fs.FS
+		ID() ID
+	}
+	// TODO: better name; OpenPOSIXFile?
+	OpenFileFS interface {
```

Comment:
The method signature for this now matches Go's standard `os.OpenFile` and isn't strictly POSIX per-say. We can leave it as is.

Reply:
Remark removed in f9a75020973f3af2b73f2889cb025e6c094904a8